symengine/functions.cpp

```diff
@@ -2895,6 +2895,116 @@ RCP<const Basic> uppergamma(const RCP<const Basic> &s, const RCP<const Basic> &x
 }
 
 
+PolyGamma::PolyGamma(const RCP<const Basic> &n, const RCP<const Basic> &x)
+    : n_{n}, x_{x}
+{
+    SYMENGINE_ASSERT(is_canonical(n_, x_))
+}
```

Comment:
I think we should move this constructor into the header file. The reason is that if it is in the header file, the compiler can inline it. Since in Release mode the only thing it does is initialize the internal instance member variables `n_` and `x_`, this becomes a very efficient operation. If it is declared in the cpp file, then the compiler must call a function. I think I've seen performance changes when I did this change in some other classes.

We should update the rest of the code to put all the constructors into the header file.